### PR TITLE
Fix regexp used to extract Urls (and fixes a whole suite of test failures)

### PR DESCRIPTION
**Pull request by DonJayamanne**

### CODE REVIEW

File: src/kernels/jupyter/launcher/jupyterConnection.node.ts [hunk 1]
```diff
@@ -142,12 +141,11 @@ export class JupyterConnectionWaiter implements IDisposable {
         }
     }
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    private getJupyterURLFromString(data: any) {
+    private getJupyterURLFromString(data: string) {
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        const urlMatch = urlMatcher.exec(data) as any;
-        const groups = urlMatch.groups() as RegExpValues.IUrlPatternGroupType;
```
Comment (DonJayamanne) [inline src/kernels/jupyter/launcher/jupyterConnection.node.ts:149]: groups is not a function, and that throws errors.

Comment (DonJayamanne) [reply src/kernels/jupyter/launcher/jupyterConnection.node.ts:149]: The compiler didn't complain as we were casting this to `any`.

File: src/kernels/jupyter/launcher/jupyterConnection.node.ts [hunk 2]
```diff
@@ -175,7 +173,7 @@ export class JupyterConnectionWaiter implements IDisposable {
     }
 
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (rebornix) [inline src/kernels/jupyter/launcher/jupyterConnection.node.ts:175]: The lint rule comment can probably be removed

File: src/kernels/jupyter/launcher/jupyterConnection.node.ts [hunk 3]
```diff
@@ -142,12 +141,11 @@ export class JupyterConnectionWaiter implements IDisposable {
         }
     }
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    private getJupyterURLFromString(data: any) {
+    private getJupyterURLFromString(data: string) {
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (rebornix) [inline src/kernels/jupyter/launcher/jupyterConnection.node.ts:145]: Same here

**Review by rebornix — APPROVED** [review]

File: src/kernels/jupyter/launcher/jupyterConnection.node.ts (hunk 2 above)
Comment (DonJayamanne) [reply src/kernels/jupyter/launcher/jupyterConnection.node.ts:175]: Oops, missed that, sory, will submit another PR

**Pull request merged**